scalafix-core/shared/src/main/scala/scalafix/internal/config/DisableConfig.scala

```diff
@@ -2,18 +2,67 @@ package scalafix
 package internal.config
 
 import metaconfig._
-import org.langmeta.Symbol
-
-import scalafix.internal.util.SymbolOps
 import metaconfig.annotation.{Description, ExampleValue}
 import metaconfig.generic.Surface
+import org.langmeta.Symbol
+
+import scala.util.matching.Regex
+
+case class DisabledSymbol(
+    @Description("Symbol to ban.")
+    symbol: Option[Symbol.Global],
+    @Description("Custom message.")
+    message: Option[String],
+    @Description("Custom id for error messages.")
+    id: Option[String],
+    @Description(
+      "When true all overrides of the symbol are also banned. Works only with symbol field, not regex. Doesn't work in unlessSynthetic mode.")
+    banHierarchy: Boolean,
+    @Description(
+      "Regex to ban instead of one symbol. When symbol field is specified, this field is ignored.")
+    regex: Option[String]) {
```

Comment:
Let's make this `Regex` so pattern errors are caught when loading rules instead of during fixing time. I think there's a `ConfDecoder[Regex]` somewhere

Reply:
Actually, this should be a `FilterMatcher` so we can support includes and excludes. You can use the `FilterMatches.matches(String): Boolean` method to test if a string matches the filter.